azure-pipelines.nuget.yml

```diff
@@ -0,0 +1,143 @@
+trigger: none
+
+parameters:
+  - name: version
+    displayName: Version to stamp on binaries and nuget package
+    type: string
+
+pool:
+  vmImage: "windows-latest"
+
+variables:
+  solution: "src/AppInstallerCLI.sln"
+  buildConfiguration: "Release"
+  packageName: Microsoft.WindowsPackageManager.Utils
+
+jobs:
+  - job: "Build"
+    variables:
+      BuildVer: $[counter(${{ parameters.version }}, 1)]
+      version: ${{ parameters.version }}.$(BuildVer)
+    steps:
+      - script: echo $(version)
+
+      - task: NuGetToolInstaller@1
+        displayName: Install Nuget
+
+      # Restores all projects, including native (vcxproj) projects
+      - task: NuGetCommand@2
+        displayName: Restore Packages
+        inputs:
+          restoreSolution: "$(solution)"
+
+      # Restores only .NET core projects, but is still necessary, as without this the IndexCreationTool and LocalhostWebServer projects fail to build
+      - task: DotNetCoreCLI@2
+        displayName: DotNet Restore
+        inputs:
+          command: "restore"
+          projects: "**/*.csproj"
+
+      - task: PowerShell@2
+        displayName: Update Binary Version
+        condition: not(eq(variables['Build.Reason'], 'PullRequest'))
+        inputs:
+          filePath: 'src\binver\Update-BinVer.ps1'
+          arguments: '-TargetFile binver\binver\version.h -BuildVersion $(BuildVer) -MajorMinorOverride ${{ parameters.version }}'
+          workingDirectory: "src"
+
+      - task: VSBuild@1
+        displayName: Build Solution x86
+        inputs:
+          platform: "x86"
+          solution: "$(solution)"
+          configuration: "$(buildConfiguration)"

```

Comment:
Is there not a way to build only the WinGetUtil (and associated dependencies)?

Reply:
I did look for that, since building the whole solution takes so long! I couldn't find anything though. I did try just building the csproj, but that failed of course, and I didn't find any parameters on the various msbuild tasks that supported specifying only a certain project and dependencies. One could hardcode a chain of build commands to build all the individual dependencies and then wingetutil but that didn't feel right.

Reply:
I believe you can specify that by setting the target to build as an argument to MSBuild. Like `msbuildArguments: "/target:WingetUtil"`